### ISSUE TITLE
DKB PIN wird nicht im Schlüsselbund gespeichert

### ISSUE THREAD

**Issue by foss-** [pecuniabanking/pecunia-client#473]
Pecunia 1.3.0 (005)
macOS 10.12

Bei Abfrage neuer Umsätze taucht jedes mal der Dialog für die DKB PIN auf. Diese gebe ich ein und setze die Checkbox zum speichern in der macOS Keychain.

Das speichern findet aber scheinbar nicht statt (oder die Abfrage aus der Keychain klappt nicht?), so dass bei jeder weiteren Umsatzabfrage erneut der identische Dialog aufscheint.

![dkb pin abfrage](https://cloud.githubusercontent.com/assets/5708172/18907265/45901f78-856b-11e6-931b-edf6141295de.jpg)

Dieses Problem setzt sich fort in einer Fehlermeldung zum Kreditkartenkonto:

![kredit](https://cloud.githubusercontent.com/assets/5708172/18908034/5396952c-856e-11e6-853c-619d560c7eeb.png)

In einigen Fällen schließt sich Pecunia bei erneutem Abruf. Allerdings wird kein Crashlog bei neustart angezeigt.

**Comment by jp112sdl** [comment]
Ähnliches Problem auch bei mir.

**Comment by richardxyx** [comment]
Exakt gleiches Problem mit gleicher Version/System.

**Comment by exislow** [comment]
Same problem here.

**Comment by mike-lischke** [comment]
Ja, das ist ein blödes Problem. Da der Abruf der Kreditkarte(n) fehlschlägt, wird die PIN dafür (die ja meist auch für andere Konten bei derselben Bank dient) nicht gespeichert, ja sogar eine gespeicherte PIN für diese Bankkennung wieder gelöscht. Das ist nervig und muss geändert werden. Fürs erste bietet es sich an die automatische Abfrage der Kreditkartenkonten abzuschalten (siehe Konteneigenschaften, z.B. via double click auf ein Konto), bis dieses Problem gelöst ist. Danach verbleibt die (nun nochmals einzugebende) PIN im Schlüsselbund.

**Comment by jp112sdl** [comment]
siehe [https://github.com/pecuniabanking/pecunia-client/issues/461]

Es gibt eine vorab Support-Version, mit der funktionieren auch die DKB Buchungen + PIN-Speicherung wieder.
[http://www.pecuniabanking.de/downloads/1.0/support/Pecunia13.dmg]

**Comment by mike-lischke** [comment]
Das Problem mit den gelöschten PINs besteht aber dennoch. Das ist ein ungewollter Seiteneffekt des Codes, der bei der Erstabfrage der PIN arbeitet und dafür sorgt, dass die PIN gelöscht wird, wenn sie nicht akzeptiert wurde.

**Comment by jp112sdl** [comment]
@mike-lischke Hmm, seltsam. Also ich hatte ursprünglich dasselbe Verhalten wie hier oben beschrieben.
VISA DKB wird nicht abgerufen und jedes Mal wurde nach dem PIN gefragt.
Hab dann die genannte Version heruntergeladen und seitdem läufts mit DKB VISA wieder tadellos und auch die PIN bleibt gespeichert.

Oder ist dein Comment jetzt so zu verstehen, dass mal generell die PIN-Verwaltung überarbeitet werden sollte, so dass diese im Fehlerfall nicht immer gelöscht wird?

**Comment by mike-lischke** [comment]
@jp112sdl Ja klar, wenn der Abruf nicht mehr fehlschlägt, dann bleibt die PIN gespeichert. Das Problem mit den Kreditkarten brachte dieses Problem erst zum Vorschein. Deshalb mein Vorschlag vorrübergehend die automatische Abfrage derselben abzustellen (manuell Abfragen wird aber dennoch die PIN wieder löschen, bis die KK wieder funktionieren).

**Comment by foss-** [comment]
Dieses Problem tritt schon länger nicht mehr auf und ist in 2.1.9 nicht mehr reproduzierbar.
